### PR TITLE
{WIP} [csl] test resync after parent is down for a long period

### PR DESCRIPTION
**Pull request by edmont**
Add some test steps to verify that a CSL Receiver is able to re-establish
CSL synchronization after its peer has been absent for a long period.

### CODE REVIEW

**Comment by edmont** [comment]
@abtink @Irving-cl, this seems to be a regression we have introduced recently, the same test passes in older revisions. More investigation is needed to see why the SSED doesn't send the last Child Update Response.

**Comment by edmont** [comment]
Could it be that `false` is returned here since the SubMac parameters didn't change: https://github.com/openthread/openthread/blob/main/src/core/mac/mac.cpp#L2278

**Comment by abtink** [comment]
@edmont I have a theory of what may be causing this:
- When device detaches, from `Mle::SetStateDetached()` we do call `Mac::UpdateCsl()`
```c++
void Mac::UpdateCsl(void)
{
    uint16_t period;
    uint8_t  channel;

    VerifyOrExit(IsCslSupported());

    period  = Get<Mle::Mle>().IsRxOnWhenIdle() ? 0 : GetCslPeriod();
    channel = GetCslChannel() ? GetCslChannel() : mRadioChannel;
```
- But now `IsCslSupported()` will return `false` and we exit without calling the `SubMac` to change CSL period (set it to zero to disable CSL)
- Later on when device attach again as child, the `SubMac` call returns `false` since nothing is changed (same period, same parent, same channel),

**Comment by abtink** [comment]
@edmont submitted https://github.com/openthread/openthread/pull/7918.

This change seems to pass the new test you added in this PR on my machine (with one small change to add some extra time for child to attach).

File: tests/scripts/thread-cert/v1_2_test_csl_transmission.py [hunk 1]
```diff
@@ -125,6 +125,22 @@ def test(self):
...
+        self.nodes[SSED_1].set_csl_timeout(8)
+        self.nodes[SSED_1].set_timeout(10)
+        self.simulator.go(2)
+        self.nodes[LEADER].stop()
+        self.simulator.go(25)
+        self.flush_all()
+        self.nodes[LEADER].start()
+        self.simulator.go(config.LEADER_STARTUP_DELAY)
+        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+        self.assertEqual(self.nodes[SSED_1].get_state(), 'child')
```
Comment (abtink) [inline tests/scripts/thread-cert/v1_2_test_csl_transmission.py:140]: In some runs (on my machine), it fails at the ` self.assertEqual(self.nodes[SSED_1].get_state(), 'child')` step.
Seems like we may be checking too early (before child reattaches). Adding some delay seems to address it:
```suggestion
        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
        self.simulator.go(5)
        self.assertEqual(self.nodes[SSED_1].get_state(), 'child')
```

Comment (edmont) [reply tests/scripts/thread-cert/v1_2_test_csl_transmission.py:140]: Sure, I'll update and rebase once 7918 is in.

**Comment by jwhui** [comment]
Merged with #7918

**Pull request closed without merging**